### PR TITLE
[CSM O11y test] Filter out time series with no valid data point

### PR DESCRIPTION
**Pull request by stanley-cheung**
- From a prod test failure, we observed that there may be time series data in a metric that contain no valid data points. It's unclear why they come to be at the moment
- This PR attempts to not count time series when there is no valid data point in the time series

Can't really do the filtering at the query level because the monitoring API does not support filtering by data points: https://cloud.google.com/monitoring/api/v3/filters

### CODE REVIEW

**Comment by stanley-cheung** [comment]
Note that this did not really address the root cause as in _why are there an extra time series with no valid data point_.

File: tests/gamma/csm_observability_test.py [hunk 1]
```diff
@@ -447,6 +450,22 @@ def query_metrics(
             results[metric] = metric_time_series
         return results
 
+    def is_legit_time_series(
+        self, series: monitoring_v3.types.TimeSeries
+    ) -> bool:
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:455]: This method doesn't really need object state, let's make it a `@classmethod`

File: tests/gamma/csm_observability_test.py [hunk 2]
```diff
@@ -447,6 +450,22 @@ def query_metrics(
...
+        for point in series.points:
+            # Test flake: we found some time series with no actual data point
+            # in prod test runs.
+            # Here we will only include time series with actual data in it.
+            if (
+                point.value.distribution_value.count > 0
+                or point.value.double_value > 0
+            ):
+                return True
+        logger.error("Warning: found time_series with no valid data point")
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:465]: `logger.warning`

File: tests/gamma/csm_observability_test.py [hunk 3]
```diff
@@ -447,6 +450,22 @@ def query_metrics(
...
+        logger.error(series)
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:466]: merge with previous logger statement

File: tests/gamma/csm_observability_test.py [hunk 4]
```diff
@@ -447,6 +450,22 @@ def query_metrics(
...
+        self, series: monitoring_v3.types.TimeSeries
+    ) -> bool:
+        for point in series.points:
+            # Test flake: we found some time series with no actual data point
+            # in prod test runs.
+            # Here we will only include time series with actual data in it.
+            if (
+                point.value.distribution_value.count > 0
+                or point.value.double_value > 0
+            ):
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:463]: Pythonistic fix :)
```suggestion
            if point.value.distribution_value.count or point.value.double_value:
                return True
```

File: tests/gamma/csm_observability_test.py [hunk 5]
```diff
@@ -429,7 +429,10 @@ def query_metrics(
                 view=monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.FULL,
                 retry=retry_settings,
             )
-            time_series = list(response)
+            time_series = []
+            for series in response:
+                if self.is_legit_time_series(series):
+                    time_series.append(series)
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:435]: Optional: can be simplified to 
```suggestion
            time_series = list(filter(self.is_legit_time_series, response))
```

**Review by sergiitk — CHANGES_REQUESTED** [review]

File: tests/gamma/csm_observability_test.py (hunk 1 above)
Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py]: Done

File: tests/gamma/csm_observability_test.py (hunk 2 above)
Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py]: Done

File: tests/gamma/csm_observability_test.py (hunk 4 above)
Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py]: Done

File: tests/gamma/csm_observability_test.py (hunk 5 above)
Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py]: Done

File: tests/gamma/csm_observability_test.py
Comment (sergiitk) [inline tests/gamma/csm_observability_test.py:1]: Resolve the conversation once the test pass to merge the PR:
- [ ] [grpc/core/master/linux/psm-csm](https://source.cloud.google.com/results/invocations/7c2ea706-8a64-4c44-af92-04c5d190d400)

**Review by sergiitk — APPROVED** [review]
All good - will enable PR auto-merge, but block on the unresolved conversation below.

File: tests/gamma/csm_observability_test.py
Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py:1]: Job passed

**Pull request merged**